crates/hardforks/src/error.rs

```diff
@@ -0,0 +1,11 @@
+#[derive(Debug)]
+
+/// Error type for hardfork related errors.
+pub struct HardforkError(pub alloc::string::String);
```

Comment:
```suggestion
pub struct HardforkError(alloc::string::String);
```

let's keep this private and instead a `fn new`